Ray_Serve/CV-Serve.ipynb

```diff
@@ -0,0 +1,1173 @@
+    "    -   Request batching\n",
+    "-   Composition Patterns\n",
+    "    -   Imperative\n",
+    "    -   Declarative / Graph Deployment API\n",
+    "-   Architecture / Under-the-hood\n",
+    "    -   Ray cluster perspective - processes / workers / actors\n",
+    "    -   Request routing, queuing, load balancing in Serve\n",
+    "\n",
+    "### Example scenario: computer vision services\n",
+    "\n",
+    "For our example use case, we’ll see how to leverage Ray Serve to host a CV segmentation\n",
+    "model and how to enhance it using additional services such as image preprocessing.\n",
+    "\n",
+    "### Context: Ray AIR\n",
+    "\n",
+    "Ray AIR is the Ray AI Runtime, a set of high-level easy-to-use APIs for\n",
+    "ingesting data, training models – including reinforcement learning\n",
+    "models – tuning those models and then serving them.\n",
+    "\n",
+    "<img src=\"https://technical-training-assets.s3.us-west-2.amazonaws.com/Introduction_to_Ray_AIR/e2e_air.png\" width=600 loading=\"lazy\"/>\n",
+    "\n",
+    "Key principles behind Ray and Ray AIR are\n",
+    "* Performance\n",
+    "* Developer experience and simplicity"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dfc41d8f-4332-4359-b957-f26bc79de7a3",
+   "metadata": {},
+   "source": [
+    "# Ray Serve\n",
+    "\n",
+    "Serve is a microservices framework for serving ML – the model serving\n",
+    "component of Ray AIR.\n",
+    "\n",
+    "<img src='https://technical-training-assets.s3.us-west-2.amazonaws.com/Ray_Serve/serve_architecture.png' width=700/>\n",
+    "\n",
+    "# Deployments\n",
+    "\n",
+    "`Deployment` is the fundamental user-facing element of serve.\n",
+    "\n",
+    "<img src='https://technical-training-assets.s3.us-west-2.amazonaws.com/Ray_Serve/deployment.png' width=600/>\n",
+    "\n",
+    "## Our First Service\n",
+    "\n",
+    "Let’s jump right in and get something simple up and running on Ray\n",
+    "Serve."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "48953d75-e69a-4fe3-adf8-079edc953683",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "@serve.deployment\n",
+    "class TextConverter:\n",
+    "    def convert(self, text):\n",
+    "        return \"***\" + str.upper(text) + \"***\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "62641bcc-f2a0-48ee-ac1d-182688552af4",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "app_handle = serve.run(TextConverter.bind())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bcf7ce21-ee2a-48cb-8043-e986a4522e89",
+   "metadata": {},
+   "source": [
+    "## Key APIs and concepts\n",
+    "\n",
+    "`Deployment` represents a service and is created with the `@serve.deployment` decorator\n",
+    "* As end users, we don't instantiate `Deployment`s directly\n",
```

Comment:
I'm not exactly sure what this means

Reply:
Typical programmers think: define class, instantiate class.

So I want to clarify that the Deployment + bind is more like a "recipe" for instantiation which Serve uses to create replicas later. We don't "prep" the Deployment constructor with bind, but don't actually call it ourselves.

Reply:
yeah that makes sense, this part "As end users, we don't instantiate `Deployment`s directly" is confusing